src/MiniProfiler/ProfileProviders/WebRequestProfilerProvider.cs

```diff
@@ -150,7 +150,23 @@ private static void EnsureName(MiniProfiler profiler, HttpRequest request)
         /// </summary>
         public override MiniProfiler GetCurrentProfiler() => Current;
 
+        /// <summary>
+        /// Current head timing.
+        /// </summary>
+        public override Timing CurrentHead
```

Comment:
I agree it's possible WebRequest provider can go away, but it's a fairly drastic behavior change and possible weight and breaking change to do such. Consider at the moment the case where there is no profiler because of the async context breaks to backend code. If suddenly there is a profiler on massive operations, that may add significant weight and allocations (even from a stack trace strings standpoint), as well as serialization costs at the end to that path. This can easily break things.

I'm not against it per se, but it'd definitely be a top breaking change alert in the v4 notes.